packages/forms-web-app/src/lib/is-token-valid.js

```diff
@@ -1,45 +1,54 @@
 const { checkToken } = require('./appeals-api-wrapper');
 const { isTokenExpired } = require('./is-token-expired');
-const { utils, enterCodeConfig } = require('@pins/common');
+const { utils } = require('@pins/common');
 const config = require('../config');
 
 const testConfirmEmailToken = '12345';
 
-const isTokenValid = async (id, token, session) => {
-	let result = {
-		valid: false,
-		action: ''
-	};
-
-	if (!id || typeof id !== 'string' || !token || typeof token !== 'string') {
-		return result;
-	}
-
-	// is test LPA + Test environment + test token
-	// then allow through as a confirm email token
-	if (
+// is test LPA + Test environment + test token
+// then allow through as a confirm email token
+const isTestLPAAndEnvironment = (token, session) => {
+	return (
 		config.server.allowTestingOverrides &&
 		utils.isTestLPA(session?.appeal?.lpaCode) &&
 		token === testConfirmEmailToken
-	) {
-		result.valid = true;
-		result.action = enterCodeConfig.actions.confirmEmail;
-		return result;
-	}
+	);
+};
 
+const getToken = async (id, token) => {
 	let tokenDocument;
 	try {
 		tokenDocument = await checkToken(id, token);
+		return tokenDocument;
 	} catch (err) {
+		console.log(err);
 		// todo: can we improve this to not rely on string matching,
 		// handler swallows response and only gives back status message
 		if (err.message === 'Too Many Requests') {
-			result.tooManyAttempts = true;
-			return result;
+			return {
+				valid: false,
+				action: '',
+				tooManyAttempts: true
+			};
 		} else {
 			throw err;
 		}
 	}
+};
+
+const isTokenValid = async (id, token) => {
+	let result = {
+		valid: false,
+		action: ''
```

Comment:
Think is is mentioned on another PR - I'm not sure if this is a breaking change versus what is in main at the moment because we aren't setting the action on the token?

Reply:
Yup that was the regression issue in 5882. Now fixed.